test/crit/main.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit"
+)
+
+func main() {
+	imgs := getImgs()
+	for _, img := range imgs {
+		log.Println("=== ", img)
+		testImg := fmt.Sprintf("%s.test.img", filepath.Base(img))
+		c := crit.New(img, testImg, "", false, false)
+		// Decode the binary image file
+		decodedImg, err := c.Decode()
+		if err != nil {
+			log.Fatal("[DECODE]: ", err)
+			continue
+		}
+		// Encode it into test binary image file
+		err = c.Encode(decodedImg)
+		if err != nil {
+			log.Fatal("[ENCODE]: ", err)
+			continue
+		}
+		// Open and compare original and test files
+		imgBytes, err := os.ReadFile(img)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		testImgBytes, err := os.ReadFile(testImg)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		if !bytes.Equal(imgBytes, testImgBytes) {
+			log.Fatal("[RECODE]: files do not match")
+		}
+		log.Println("=== PASS")
+	}
+}
+
+func getImgs() []string {
+	var skipImgs = []string{
+		"pages-",
+		"iptables-",
+		"ip6tables-",
+		"nftables-",
+		"route-",
+		"route6-",
+		"ifaddr-",
+		"tmpfs-",
+		"netns-ct-",
+		"netns-exp-",
+		"rule-",
+	}
+	var imgs []string
+	dir, err := os.ReadDir(".")
```

Comment:
Maybe we could use [`filepath.Walk()`](https://pkg.go.dev/path/filepath#WalkDir)?
```go
func getImgs() []string {
	var skipImgs = []string{
		"pages-",
		"iptables-",
		"ip6tables-",
		"nftables-",
		"route-",
		"route6-",
		"ifaddr-",
		"tmpfs-",
		"netns-ct-",
		"netns-exp-",
		"rule-",
	}
	var imgs []string

	re := regexp.MustCompile(`.*\.img`)
	fn := func(path string, info os.DirEntry, err error) error {
		switch {
		case err != nil:
			return err
		case !info.Type().IsRegular():
			return nil
		case !re.MatchString(info.Name()):
			return nil
		}

		imgName := info.Name()
		for _, skip := range skipImgs {
			if strings.HasPrefix(imgName, skip) {
				return nil
			}
		}

		imgs = append(imgs, path)
		return nil
	}

	err := filepath.WalkDir(".", fn)
	if err != nil {
		log.Fatal(err)
	}

	return imgs
}
```

Reply:
Since we know that all the generated images will be present at the top-level directory, `filepath.WalkDir()` seems like overkill. `os.ReadDir()` is also faster, so we can leverage that as we don't need to recurse subdirectories.